src/wiringx.c

```diff
@@ -103,6 +86,142 @@ static struct spi_t spi[2] = {
 	} while (0)
 #endif
 
+#ifndef __FreeBSD__
+/* SPI Bus Parameters */
+struct spi_t {
+	uint8_t mode;
+	uint8_t bits_per_word;
+	uint16_t delay;
+	uint32_t speed;
+	int fd;
+} spi_t;
+
+/* SPI states list implementation */
+
+struct spi_id_t {
+	uint8_t device;
+	uint8_t channel;
+};
+
+struct spi_list_t {
+	struct spi_id_t id;
+	struct spi_t data;
+};
+
+struct spi_list_t *spilist = NULL;
+int nrspilist = 0;
```

Comment:
Can you add `static` in the right spots like here.

Reply:
static sounds like a hack to me. The variable should only exist in one place, and if it is declared anywhere else its gotta be labeled extern

Reply:
>This is a local variable with global lifetime.

`static` doesn't do anything with the lifetime of a variable. It only affects the scope. `static` limits to scope to the source file only. That's exactly what we want.

> Lets keep it in a way that people can use all the names they like inside other source files.

If people want to use a `nrspilist` in another source file they can't because it will conflict with the non-static nrspilist in this source file. And, access local variable shouldn't be done without using functions that allow you to touch them.

Reply:
I guess I misunderstood the meaning of static ... if that is what static does, limit the scope to this file, its good.
I thought it told the compiler that there is only one instance of this name.

Reply:
A good thing we have wikipedia:

> In the C programming language, static is used with global variables and functions to set their scope to the containing file.